drivers/can/can_renesas_ra.c

```diff
@@ -0,0 +1,1114 @@
+#define _ELC_EVENT_CAN_TX(channel)     ELC_EVENT_CAN##channel##_TX
+#define _ELC_EVENT_CAN_CHERR(channel)  ELC_EVENT_CAN##channel##_CHERR
+
+#define ELC_EVENT_CAN_COMFRX(channel) _ELC_EVENT_CAN_COMFRX(channel)
+#define ELC_EVENT_CAN_TX(channel)     _ELC_EVENT_CAN_TX(channel)
+#define ELC_EVENT_CAN_CHERR(channel)  _ELC_EVENT_CAN_CHERR(channel)
+
+#define CAN_RENESAS_RA_CHANNEL_IRQ_INIT(index)                                                     \
+	{                                                                                          \
+		R_ICU->IELSR_b[DT_INST_IRQ_BY_NAME(index, rx, irq)].IELS =                         \
+			ELC_EVENT_CAN_COMFRX(DT_INST_PROP(index, channel));                        \
+		R_ICU->IELSR_b[DT_INST_IRQ_BY_NAME(index, tx, irq)].IELS =                         \
+			ELC_EVENT_CAN_TX(DT_INST_PROP(index, channel));                            \
+		R_ICU->IELSR_b[DT_INST_IRQ_BY_NAME(index, err, irq)].IELS =                        \
+			ELC_EVENT_CAN_CHERR(DT_INST_PROP(index, channel));                         \
+                                                                                                   \
+		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(index, rx, irq),                                   \
+			    DT_INST_IRQ_BY_NAME(index, rx, priority), canfd_common_fifo_rx_isr,    \
+			    NULL, 0);                                                              \
+		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(index, tx, irq),                                   \
+			    DT_INST_IRQ_BY_NAME(index, tx, priority), canfd_channel_tx_isr, NULL,  \
+			    0);                                                                    \
+		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(index, err, irq),                                  \
+			    DT_INST_IRQ_BY_NAME(index, err, priority), canfd_error_isr, NULL, 0);  \
+                                                                                                   \
+		irq_enable(DT_INST_IRQ_BY_NAME(index, rx, irq));                                   \
+		irq_enable(DT_INST_IRQ_BY_NAME(index, tx, irq));                                   \
+		irq_enable(DT_INST_IRQ_BY_NAME(index, err, irq));                                  \
+	}
+
+#define CAN_RENESAS_RA_INIT(index)                                                                 \
+	PINCTRL_DT_INST_DEFINE(index);                                                             \
+	static canfd_afl_entry_t canfd_afl##index[DT_INST_PROP(index, rx_max_filters)];            \
+	struct can_renesas_ra_filter                                                               \
+		can_renesas_ra_rx_filter##index[DT_INST_PROP(index, rx_max_filters)];              \
+	static can_bit_timing_cfg_t g_canfd_bit_timing##index;                                     \
+	static const struct can_renesas_ra_cfg can_renesas_ra_cfg##index = {                       \
+		.common = CAN_DT_DRIVER_CONFIG_INST_GET(index, 500, 5000000),                      \
```

Comment:
Are these bitrate limits correct? 500 bit/s to 5Mbit/s?

Reply:
This value was updated to 0-5Mbps